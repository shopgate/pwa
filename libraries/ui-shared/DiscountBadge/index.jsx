import React from 'react';
import PropTypes from 'prop-types';
import { I18n } from '@shopgate/engage/components';
import { i18n } from '@shopgate/engage/core';
import styles from './style';

/**
 * The discount badge component.
 * @param {Object} props The component props
 * @param {string} [props.className] Additional CSS style definitions
 * @param {string} props.text The text contents of the component.
 * @returns {JSX.Element}
 */
const DiscountBadge = ({
  text,
  className,
  display,
  discount,
}) => (
  <div
    data-test-id={text}
    className="ui-shared__discount-badge"
    aria-label={`${i18n.text('cart.discount')}: ${text}`}
    tabIndex={-1}
  >
    <I18n.Text
      className={`${styles[display]} ${className}`}
      string={text}
      params={[discount]}
      aria-hidden
    />
  </div>
);

DiscountBadge.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
  discount: PropTypes.number,
  display: PropTypes.oneOf(Object.keys(styles)),
};

DiscountBadge.defaultProps = {
  className: '',
  discount: null,
  display: 'small',
};

export default DiscountBadge;
