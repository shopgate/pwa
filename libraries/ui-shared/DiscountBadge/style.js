import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const badge = {
  background: themeConfig.colors.primary,
  borderRadius: 2,
  color: themeConfig.colors.primaryContrast,
  paddingTop: 5,
  paddingBottom: 5,
  width: '100%',
  fontWeight: 700,
  textAlign: 'center',
  display: 'inline-block',
  lineHeight: 1,
  ...themeConfig.variables.discountBadgeBase,
};

/**
 * The discount badge styles that can be selected by passing the style key.
 * @type {Object}
 */
export default {
  small: css({
    ...badge,
  }).toString(),
  big: css({
    ...badge,
    paddingTop: 7,
    paddingLeft: 5,
    paddingRight: 5,
  }).toString(),
};
