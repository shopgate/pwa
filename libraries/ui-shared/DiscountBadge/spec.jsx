import React from 'react';
import { shallow } from 'enzyme';
import DiscountBadge from './index';

jest.mock('@shopgate/engage/components');

describe('<DiscountBadge />', () => {
  it('should render the text', () => {
    const wrapper = shallow(<DiscountBadge text="foo" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the text and discount', () => {
    const wrapper = shallow(<DiscountBadge text="SAVE {0}%" discount={20} />);

    expect(wrapper).toMatchSnapshot();
  });
});
