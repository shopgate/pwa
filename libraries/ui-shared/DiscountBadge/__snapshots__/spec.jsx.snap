// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscountBadge /> should render the text 1`] = `
<div
  className="ui-shared__discount-badge"
  data-test-id="foo"
>
  <Translate
    className="css-13de4di "
    params={
      Array [
        null,
      ]
    }
    role={null}
    string="foo"
    transform={null}
  />
</div>
`;

exports[`<DiscountBadge /> should render the text and discount 1`] = `
<div
  className="ui-shared__discount-badge"
  data-test-id="SAVE {0}%"
>
  <Translate
    className="css-13de4di "
    params={
      Array [
        20,
      ]
    }
    role={null}
    string="SAVE {0}%"
    transform={null}
  />
</div>
`;
